packages/vats/src/vat-bridge.js

```diff
@@ -0,0 +1,49 @@
+// @ts-check
+import { Fail } from '@agoric/assert';
+import { Far } from '@endo/far';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { zoneChainStorageNode } from '@agoric/internal/src/lib-chainStorage.js';
+import * as cb from '@agoric/internal/src/callback.js';
+import { zoneBridgeManager } from './bridge.js';
+
+export function buildRootObject(vatPowers, _args, baggage) {
+  const { D } = vatPowers;
+  D || Fail`D missing in vatPowers ${Object.keys(vatPowers)}`;
+
+  const zone = makeDurableZone(baggage);
+
+  const provideManagerForBridge = zoneBridgeManager(
+    zone.subZone('BridgeManager'),
+    D,
+  );
+  const makeChainStorageNode = zoneChainStorageNode(
+    zone.subZone('ChainStorageNode'),
+  );
```

Comment:
Given that `zoneBridgeManager` starts with `zone.subZone('ScopedManager').exoClass('BridgeScopedManager', …)`, `zone.exoClassKit('BridgeManagerKit', …)`, and `zone.mapStore('bridgeToManagerKit')`; and that `zoneChainStorageNode` starts with `zone.exoClass('ChainStorageNode', …)`, isn't this redundant nesting? I'd much rather see an arrangement in which all zone keys appear at the same high level. If we're not comfortable with representing specific MapStore keys as objects, perhaps instead we need a way to attenuate the zones?
```suggestion
  const view = zone.attenuate({
    keys: ['BridgeScopedManager', 'BridgeManagerKit', 'bridgeToManagerKit'],
  });
  const provideManagerForBridge = zoneBridgeManager(view, D);
  const makeChainStorageNode = zoneChainStorageNode(view);
```

Reply:
The mechanism for attenuating the zones is to create a subzone.  Attenuation based on filtering keys by name is discouraged because it is tightly coupled to the library using the zone, for which the specific key names are an implementation detail.

Your critique about redundant nesting for the `subZone('ScopedManager')` is a good one, though.  I'll fix that.

Reply:
> Attenuation based on filtering keys by name is discouraged because it is tightly coupled to the library using the zone, for which the specific key names are an implementation detail.

Understood, but this cure seems worse than the disease. If I understand correctly, we're going to end up with baggage full of deeply nested but mostly empty maps, which will be unnecessarily difficult to debug:
```
MapStore{
  BridgeManager: MapStore{
    ScopedManager: MapStore{
      BridgeScopedManager_kindHandle: Far('kind', {}),
    },
    BridgeManagerKit_kindHandle: Far('kind', {}),
    bridgeToManagerKit: MapStore{…},
  },
  ChainStorageNode: MapStore{
    ChainStorageNode_kindHandle: Far('kind', {}),
  },
  …
}
```
while I think we should strive for something more like
```
MapStore{
  BridgeChannel_kindHandle: Far('kind', {}),
  BridgeManagerKit_kindHandle: Far('kind', {}),
  bridgeToManagerKit: MapStore{…},
  ChainStorageNode_kindHandle: Far('kind', {}),
  …
}
```

Reply:
Just a minor correction: I've already changed this so that there is no `ScopedManager MapStore`, which leaves the general structure 2 levels deep (first is controlled by the vat/contract initialization code with keys corresponding to packages, second is keys controlled by the individual package).  Using `subZone`s in this way is a straightforward (once documented) way to protect against one package stealing keys from another package.

If having 2 levels is still too difficult to debug, it is worth mentioning that the concept of a `subZone` doesn't inherently need to be implemented as deeply-nested maps.  Another way forward could be to use a flat map with key prefixes to represent deep keys, somewhat like how you did with `chainStorageNode`.

```
MapStore{
  "BridgeManager:BridgeManagerKit_kindHandle": Far('kind', {}),
  "BridgeManager:BridgeScopedManager_kindHandle": Far('kind', {}),
  "BridgeManager:bridgeToManagerKit": MapStore{...},
  "ChainStorageNode:ChainStorageNode_kindHandle": Far('kind', {}),
  ...
}
```

Is either of these alternatives acceptable to you?  Do they inspire you to propose something else?

Reply:
I do not understand the problem 
    * with nested sub-zones as attenuation. If another mechanism would not replace sub-zones, then it would seem to be unnecessary additional complexity. And I can't see how it could replace sub-zones.
    * with nested sub-zones resulting in nested baggage map-stores. Seems more straightforward than imposing a prefixing convention on flat names. Also, any string mangling system would need to worry about quoting confusion and hilbert, which we simply avoid by using nested maps.

We should probably have a verbal discussion.

Reply:
@erights We should keep in mind that the current state of sub-zones is insufficient to handle things that will definitely be needed, like limits for key counts, key length, entry depth, value size, and total data usage, and possibly  even for things like key pattern. We can probably add all of that to sub-zones, but they may not turn out to be the right abstraction (for example, in a filesystem, directories are generally quite distinct from regular files and sometimes control is delegated at the level of a directory but other times it is at the level of one or more specific files—there's a reason why e.g. [`mktemp`](https://www.man7.org/linux/man-pages/man1/mktemp.1.html) supports a `--directory` option rather than _always_ creating a directory, even though single-file behavior is possible to implement on top of a directory-only foundation).